elliptic-curve/src/point/non_identity.rs

```diff
@@ -103,6 +108,81 @@ impl<P> AsRef<P> for NonIdentity<P> {
     }
 }
 
+impl<const N: usize, P> BatchNormalize<[Self; N]> for NonIdentity<P>
+where
+    P: Curve + BatchNormalize<[P; N], Output = [P::AffineRepr; N]>,
+{
+    type Output = [NonIdentity<P::AffineRepr>; N];
+
+    fn batch_normalize(points: &[Self; N]) -> [NonIdentity<P::AffineRepr>; N] {
+        // Ensure casting is safe.
+        // This always succeeds because `NonIdentity` is `repr(transparent)`.
+        debug_assert_eq!(size_of::<P>(), size_of::<NonIdentity<P>>());
+        debug_assert_eq!(align_of::<P>(), align_of::<NonIdentity<P>>());
+
+        #[allow(unsafe_code)]
+        // SAFETY: `NonIdentity` is `repr(transparent)`.
+        let points: &[P] = unsafe { slice::from_raw_parts(points.as_ptr().cast(), N) };
+        let points = points.try_into().expect("slice should be size `N`");
+        let affine_points = <P as BatchNormalize<_>>::batch_normalize(points);
+
+        // Ensure `array::map()` can be optimized to a `memcpy`.
+        debug_assert_eq!(
+            size_of::<P::AffineRepr>(),
+            size_of::<NonIdentity<P::AffineRepr>>()
+        );
+        debug_assert_eq!(
+            align_of::<P::AffineRepr>(),
+            align_of::<NonIdentity<P::AffineRepr>>()
+        );
+
+        affine_points.map(|point| NonIdentity { point })
```

Comment:
My comment https://github.com/RustCrypto/traits/pull/1896/files#r2142491767 was supposed to be about this line, apologies.

Here, the best alternative to `mem::transmute_copy()`, is `ptr::read()`.

However, as noted before, `array::map()` seems to optimize just fine.